test/integration/noderesourcetopology_test.go

```diff
@@ -555,6 +559,286 @@ func TestTopologyMatchPlugin(t *testing.T) {
+			pods: []*v1.Pod{
+				withReqAndLimit(
+					withReqAndLimit(
+						st.MakePod().Namespace(ns).Name("topology-aware-scheduler-pod"),
+						map[v1.ResourceName]string{v1.ResourceCPU: "3", v1.ResourceMemory: "5Gi"}),
+					map[v1.ResourceName]string{v1.ResourceCPU: "3", v1.ResourceMemory: "5Gi"}).Obj(),
+			},
+			nodeResourceTopologies: []*topologyv1alpha1.NodeResourceTopology{
+				makeNodeResourceTopology("fake-node-1", topologyv1alpha1.SingleNUMANodeContainerLevel,
+					[]topologyv1alpha1.ResourceInfoList{
+						{
+							noderesourcetopology.MakeTopologyResInfo(cpu, "2", "2"),
+							noderesourcetopology.MakeTopologyResInfo(memory, "8Gi", "8Gi"),
+						},
+						{
+							noderesourcetopology.MakeTopologyResInfo(cpu, "2", "2"),
+							noderesourcetopology.MakeTopologyResInfo(memory, "8Gi", "8Gi"),
+						},
+					}),
+				makeNodeResourceTopology("fake-node-2", topologyv1alpha1.SingleNUMANodeContainerLevel,
+					[]topologyv1alpha1.ResourceInfoList{
+						{
+							noderesourcetopology.MakeTopologyResInfo(cpu, "4", "4"),
+							noderesourcetopology.MakeTopologyResInfo(memory, "8Gi", "8Gi"),
+						},
+						{
+							noderesourcetopology.MakeTopologyResInfo(cpu, "4", "4"),
+							noderesourcetopology.MakeTopologyResInfo(memory, "8Gi", "8Gi"),
+						},
+					}),
+			},
+			expectedNodes: []string{"fake-node-2"},
+		},
+		{
+			name: "SingleNUMANodeContainerLevel: Filtering out nodes that cannot fit resources in case of Guaranteed pod with init container",
+			pods: []*v1.Pod{
+				initWithReqAndLimit(
+					withReqAndLimit(
+						st.MakePod().Namespace(ns).Name("topology-aware-scheduler-pod"),
+						map[v1.ResourceName]string{v1.ResourceCPU: "2", v1.ResourceMemory: "4Gi"}),
+					map[v1.ResourceName]string{v1.ResourceCPU: "4", v1.ResourceMemory: "4Gi"}).Obj(),
+			},
+			nodeResourceTopologies: []*topologyv1alpha1.NodeResourceTopology{
+				makeNodeResourceTopology("fake-node-1", topologyv1alpha1.SingleNUMANodeContainerLevel,
+					[]topologyv1alpha1.ResourceInfoList{
+						{
+							noderesourcetopology.MakeTopologyResInfo(cpu, "2", "2"),
+							noderesourcetopology.MakeTopologyResInfo(memory, "8Gi", "8Gi"),
+						},
+						{
+							noderesourcetopology.MakeTopologyResInfo(cpu, "2", "2"),
+							noderesourcetopology.MakeTopologyResInfo(memory, "8Gi", "8Gi"),
+						},
+					}),
+				makeNodeResourceTopology("fake-node-2", topologyv1alpha1.SingleNUMANodeContainerLevel,
+					[]topologyv1alpha1.ResourceInfoList{
+						{
+							noderesourcetopology.MakeTopologyResInfo(cpu, "4", "4"),
+							noderesourcetopology.MakeTopologyResInfo(memory, "8Gi", "8Gi"),
+						},
+						{
+							noderesourcetopology.MakeTopologyResInfo(cpu, "4", "4"),
+							noderesourcetopology.MakeTopologyResInfo(memory, "8Gi", "8Gi"),
+						},
+					}),
+			},
+			expectedNodes: []string{"fake-node-2"},
+		},
+		{
+			name: "SingleNUMANodeContainerLevel: Cannot fit resources in case of Guaranteed pod with multi containers",
```

Comment:
same as above - if possible. We should try to have tests on which `sum(R) < Node` - if this is not possible, or if we can exercise a corner case, we can do otherwise, but let's add comments to explain why we add this exception.

Reply:
Done!